src/docs/references/page-schema/index.md

```diff
@@ -0,0 +1,1095 @@
+  [defs in JSON schema](https://json-schema.org/understanding-json-schema/structuring.html#defs)
+
+For instance, a page section's call to action has an optional internal
+button link. In the schema, the _fieldType_ attribute determines whether
+the section has a call to action, and whether it is an internal link or
+an external link.
+
+The schema then determines that if _fieldType_ is required and is
+defined as internal button link, the relevant additional attributes are
+added to _callToAction_ .
+
+```json
+{
+  "callToAction": {
+    "type": "object",
+    "properties": {
+      "fieldType": {
+        "title": "Section call to action",
+        "type": "string",
+        "default": "none",
+        "description": "The action the user is prompted to take after viewing the section.",
+        "oneOf": [
+          {
+            "$ref": "#/$defs/fieldType/none",
+            "title": "No call to action"
+          },
+          {
+            "$ref": "#/$defs/fieldType/internalButtonLink",
+            "title": "Internal link",
+            "description": "Link to a page in your marketplace. Displayed as a button."
+          },
+          {
+            "$ref": "#/$defs/fieldType/externalButtonLink",
+            "title": "External link",
+            "description": "Link to a page outside your marketplace. Opens in a new tab. Displayed as a button."
+          }
+        ]
+      }
+    },
+    "allOf": [
+      {
+        "if": {
+          "properties": {
+            "fieldType": {
+              "$ref": "#/$defs/fieldType/internalButtonLink"
+            }
+          },
+          "required": ["fieldType"]
+        },
+        "then": {
+          "$ref": "#/$defs/internalButtonLink"
+        }
+      }
+    ]
+  }
+}
+```
+
+The definitions for _fieldType/internalButtonLink_ and
+_internalButtonLink_ can be found in _\$defs_.
+
+```json
+{
+"$defs": {
+  "fieldType": {
+    "none": {
+      "const": "none"
+    },
+    "internalButtonLink": {
+      "const": "internalButtonLink"
+    }
+  },
+  "internalButtonLink": {
+    "properties": {
+      "content": {
+        "title": "Internal link text",
+        "type": "string",
+        "minLength": 1
+      },
+      "href": {
+        "description": "Include only the path after your domain. For example, if you want to link to your About page, use /p/about.",
+        "title": "Internal link address",
+        "type": "string",
+        "examples": [
+          "#section-id-as-anchor",
+          "/absolute/path/to/page"
+        ],
+        "pattern": "^(?![a-zA-Z][a-zA-Z\\+\\.\\-]*\\:)",
+        "minLength": 1
+      }
+    },
+    "required": [
+      "content",
+      "href"
+    ]
+  }
+}
+```
+
+For a page asset section where _callToAction.fieldType_ is internal
+button link, _callToAction_ will also have the properties defined in
+_internalButtonLink_.
+
+![Call to action with internal button](./internal-button-cta.png)
+
+## Using page asset schema when building a client
```

Comment:
I wonder if we should emphasize that the given schema is the current built-in one for Pages, but can be subject to change. I doubt that we'd want to make any breaking changes to it from now on (since custom client apps could break), but we might introduce new fields / properties, I guess.

Reply:
Added a note about this